File: TimesTables/app/src/main/java/com/timestables/MainActivity.java
package com.timestables;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.SeekBar;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ListView timeTableListView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final SeekBar timeTableSeekBar = findViewById(R.id.timesTableSeekBar);
        timeTableListView = findViewById(R.id.timesTableListView);

        timeTableSeekBar.setMax(20);
        timeTableSeekBar.setProgress(10);

        timeTableSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                int min = 1;
                int timesTable;
                if (progress < 1) {
                    timesTable = min;
                    timeTableSeekBar.setProgress(min);
                } else {
                    timesTable = progress;
                }

                Log.i("SeekBar value", Integer.toString(timesTable));

                generateTimesTable(timesTable);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });

     //   generateTimesTable(10);
    }

    public void generateTimesTable(int timesTable) {
        List<String> content = new ArrayList<>();
        for (int i = 1; i <= 10; i++) {
            content.add(i * timesTable + "");
        }

        ArrayAdapter<String> adapter = new ArrayAdapter<>
                (this, android.R.layout.simple_list_item_1, content);
        timeTableListView.setAdapter(adapter);
    }
}

File: Kotlin Demo Projects/TipCalculator/app/src/main/java/com/tipcalculator/MainActivity.kt
package com.tipcalculator

import android.os.Bundle
import android.widget.SeekBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        percentSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                if (progress == 0) {
                    percentSeekBar.progress = 1
                    tipPercentTextView.text = "1%"
                } else {
                    tipPercentTextView.text = "$progress%"
                }
                calculate()
            }

            override fun onStartTrackingTouch(seekBar: SeekBar?) {

            }

            override fun onStopTrackingTouch(seekBar: SeekBar?) {

            }
        })

        numberPeopleSeekBar.progress = 1

        numberPeopleSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
                if (progress == 0) {
                    numberPeopleSeekBar.progress = 1
                } else {
                    numberPeopleTextView.text = "$progress"
                }
                calculate()
            }

            override fun onStartTrackingTouch(seekBar: SeekBar?) {
            }

            override fun onStopTrackingTouch(seekBar: SeekBar?) {
            }


        })

        calculateButton.setOnClickListener {
            calculate()
        }
    }

    fun calculate() {
        if (billEditText.length() == 0) {
            Toast.makeText(this, "Enter a bill amount", Toast.LENGTH_LONG).show()
        } else {
            val billAmount = billEditText.text.toString().toInt()
            val tipPercent = percentSeekBar.progress
            val tipAmount = (billAmount * tipPercent) / 100
            val totalPayment = billAmount + tipAmount
            val numberOfPeople = numberPeopleSeekBar.progress
            val eachPersonShare = totalPayment / numberOfPeople

            totalPaymentTextView.text =
                "Total $${totalPayment} (Tip $${tipAmount}) ($${eachPersonShare} each)"
        }

    }
}